atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/StructureHolder.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api.cache;
+
+import java.util.function.Function;
+
+/**
+ * Utility class to make manipulating {@link io.vavr.collection.Map} and {@link io.vavr.collection.Set} less
+ * error-prone as all methods that modify state return a *new* instance in vavr.
+ */
+final class StructureHolder<V> {
+    private final V initialValue;
```

Comment:
nit/unactionable?: not wrong, though it does feel more conceptually intuitive to me to have a supplier of the initial value

Reply:
I agree, actually. This was copied from internal code.